### PR TITLE
Get parameter list

### PR DESCRIPTION
**Pull request by WalterBright**
Replaces the function `getParameters()` with `getParameterList()`

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]
Otherwise LGTM

File: src/dmd/func.d [hunk 1]
```diff
@@ -2275,21 +2275,20 @@ extern (C++) class FuncDeclaration : Declaration
      * Return the function's parameter list, and whether
      * it is variadic or not.
      */
-    final Parameters* getParameters(int* pvarargs)
+    final ParameterList getParameterList(int* pvarargs)
```
Comment (jacob-carlborg) [inline src/dmd/func.d]: Needs to update the corresponding header file.

**Review by jacob-carlborg — DISMISSED** [review]

File: src/dmd/dtemplate.d [hunk 2]
```diff
@@ -1260,7 +1260,7 @@ extern (C++) final class TemplateDeclaration : ScopeDsymbol
                  */
                 for (fptupindex = 0; fptupindex < nfparams; fptupindex++)
                 {
-                    Parameter fparam = (*fparameters)[fptupindex];
+                    auto fparam = (*fparameters.parameters)[fptupindex]; // fparameters[fptupindex] ?
```
[the remark this answers is not in the document]
Comment (WalterBright) [reply src/dmd/dtemplate.d]: Here it's indexing parameters, not fparameters.

File: src/dmd/func.d (hunk 1 above)
Comment (WalterBright) [reply src/dmd/func.d]: Is it even used by the C++ code? @ibuclaw @kinke ?

Comment (jacob-carlborg) [reply src/dmd/func.d]: The function exists in a header file, not sure if it's used. It's now out of sync.

Comment (kinke) [reply src/dmd/func.d]: Looks like LDC doesn't use it, but if it's in the header, it should be adapted too.

Comment (WalterBright) [reply src/dmd/func.d]: updated .h files

**Pull request merged**